File: CoreJava/SOLID-princples-app/src/isp/violation/Manager.java
package isp.violation;

public class Manager implements IWorker {

	@Override
	public void startsWork() {
		System.out.println("Manager starts working");
	}

	@Override
	public void endsWork() {
		System.out.println("Manager ends working");
	}

	@Override
	public void startsEat() {
		System.out.println("Manager starts eating");
	}

	@Override
	public void endsEat() {
		System.out.println("Manager stops eating");
	}

}

File: DotNET/Solid Principles/ISPViolationApp/ISPViolationApp/Manager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ISPViolationApp
{
    class Manager : IWorkable
    {
        public void EndsEat()
        {
            Console.WriteLine("Manager Stops Eating");
        }

        public void EndsWork()
        {
            Console.WriteLine("Manager Ends Working");
        }

        public void StartsEat()
        {
            Console.WriteLine("Manager Starts Eating");
        }

        public void StartsWork()
        {
            Console.WriteLine("Manager Starts Working");
        }
    }
}
